pandas/tests/indexing/multiindex/test_slice.py

```diff
@@ -492,6 +492,29 @@ def test_loc_axis_arguments(self):
         with pytest.raises(ValueError):
             df.loc(axis="foo")[:, :, ["C1", "C3"]]
 
+    def test_loc_axis_single_level_indexer(self):
+
+        # test single level indexing on Multindex column dataframe
+        df1 = pd.DataFrame(
+            np.arange(27).reshape(3, 9),
+            columns=pd.MultiIndex.from_product(
+                [["a1", "a2", "a3"], ["b1", "b2", "b3"]]
+            ),
+        )
+        result = df1.loc(axis=1)["a1":"a2"]
+        expected = df1.iloc[:, :-3]
+        tm.assert_frame_equal(result, expected)
+        result = df1.loc(axis=1)["a1"]
+        expected = df1.iloc[:, :3]
+        expected.columns = ["b1", "b2", "b3"]
+        tm.assert_frame_equal(result, expected)
+
+        # test single level indexing on single index column data frame
+        df2 = pd.DataFrame(np.arange(9).reshape(3, 3), columns=["a", "b", "c"])
+        result = df2.loc(axis=1)["a"]
+        expected = pd.Series([0, 3, 6], name="a")
+        tm.assert_series_equal(result, expected, check_dtype=False)
```

Comment:
Why is `check_dtype=False` set?

Reply:
@mroeschke  Since CI was failing for some of the 32 bit OS builds because of int32 vs int64 dtype comparison - don't know if there is any other work around.

```
================================== FAILURES ===================================
__________ TestMultiIndexSlicers.test_loc_axis_single_level_indexer ___________
[gw0] win32 -- Python 3.6.7 C:\Miniconda\envs\pandas-dev\python.exe

self = <pandas.tests.indexing.multiindex.test_slice.TestMultiIndexSlicers object at 0x00000156C517C898>

    def test_loc_axis_single_level_indexer(self):
    
        # test single level indexing on Multindex column dataframe
        df1 = pd.DataFrame(
            np.arange(27).reshape(3, 9),
            columns=pd.MultiIndex.from_product(
                [["a1", "a2", "a3"], ["b1", "b2", "b3"]]
            ),
        )
        result = df1.loc(axis=1)["a1":"a2"]
        expected = df1.iloc[:, :-3]
        tm.assert_frame_equal(result, expected)
        result = df1.loc(axis=1)["a1"]
        expected = df1.iloc[:, :3]
        expected.columns = ["b1", "b2", "b3"]
        tm.assert_frame_equal(result, expected)
    
        # test single level indexing on single index column data frame
        df2 = pd.DataFrame(np.arange(9).reshape(3, 3), columns=["a", "b", "c"])
        result = df2.loc(axis=1)["a"]
        expected = pd.Series([0, 3, 6], dtype="int64", name="a")
>       tm.assert_series_equal(result, expected)
E       AssertionError: Attributes of Series are different
E       
E       Attribute "dtype" are different
E       [left]:  int32
E       [right]: int64

pandas\tests\indexing\multiindex\test_slice.py:516: AssertionError


```

Reply:
I see you're not assigning `dtype="int64"` anymore. Could you remove `check_dtype=False` now?